sql-plugin/src/main/scala/com/nvidia/spark/rapids/basicPhysicalOperators.scala

```diff
@@ -488,20 +527,15 @@ object GpuFilter {
     }
   }
 
-  def filterAndClose(batch: ColumnarBatch,
-      boundCondition: Expression): ColumnarBatch = {
-    if (!boundCondition.deterministic) {
-      // If the condition is non-deterministic we cannot retry it, we could retry the filter, but
-      // this should be super rare. So we are not going to spend time trying to make it happen.
-      withResource(batch) { batch =>
-        GpuFilter(batch, boundCondition)
-      }
-    } else {
-      val sb = SpillableColumnarBatch(batch, SpillPriorities.ACTIVE_ON_DECK_PRIORITY)
-      RmmRapidsRetryIterator.withRetryNoSplit(sb) { sb =>
-        withResource(sb.getColumnarBatch()) { cb =>
-          GpuFilter(cb, boundCondition)
-        }
+  private def computeCheckedFilterMask(boundCondition: GpuTieredProject,
+      cb: ColumnarBatch): Option[cudf.ColumnVector] = {
+    withResource(boundCondition.project(cb)) { filterBatch =>
```

Comment:
This not using `projectSingle` because there isn't one for tiered project I imagine. Might be nice to add one in a different PR

Reply:
Also, I can't think of a case where the projection would result in 0 columns, but bringing it up in case we need to be defensive.

Reply:
There are cases when it could COUNT is one where all we have is rows, no columns. I could add in a check to verify that there is one and only one output column. I think it would be `boundExprs.tail.length == 1`, but I didn't want to spend a ton of time on it.